### DISCUSSION TITLE
Gitlab OAuth: how to configure gitlab roles (depending on `/api/v4/groups` api) <-> with grafana roles using JSMEPath ?

### DISCUSSION THREAD

**Discussion by copolycube** [grafana/grafana#53495] [Platform]
Add another RPC call when performing OAuth identification against gitlab.com in order to be able to create advanced role mapping, while the current setting only allows mapping gitlab admin into grafana admins.

* currently used: `https://gitlab.com/api/v4/users` (doc : https://docs.gitlab.com/ee/api/users.html#list-current-user-for-normal-users ) 
* feature request, add: `https://gitlab.com/api/v4/groups/:id/members`  (doc: https://docs.gitlab.com/ee/api/members.html ) and `/projects/:id/members` (same format)

In particular, it provides the following field, where the number (10, 20, 30, 40, 50) represent a gitlab role among the queried group.
eg.
```
"access_level": 30,
```

The goal is to allow more elaborate JSMEPath to map gitlab roles with grafana roles to be able to configure `GF_AUTH_GITLAB_ROLE_ATTRIBUTE_PATH` or `role_attribute_path` in a more precise way:

```
The access levels are defined in the Gitlab::Access module. Currently, these levels are recognized:

No access (0)
Minimal access (5) ([Introduced](https://gitlab.com/gitlab-org/gitlab/-/issues/220203) in GitLab 13.5.)
Guest (10)
Reporter (20)
Developer (30)
Maintainer (40)
Owner (50). Valid for projects in [GitLab 14.9 and later](https://gitlab.com/gitlab-org/gitlab/-/issues/21432).
```
 

Current doc : 
* https://grafana.com/docs/grafana/next/setup-grafana/configure-security/configure-authentication/gitlab/#map-roles : current possible mapping only detects "is admin" with no further consideration, which is a bit limited for advanced use-case.

Configuration attempts :
* https://community.grafana.com/t/oauth2-gitlab-roles-matching-jsmepath/69526

Thanks in advance !

**Comment by copolycube** [comment]
(Not sure if this should be in the "Platform" or "general" section ?)

**Comment by copolycube** [comment]
The goal is to be able to use information from the already existing setting `allowed_groups` into the JSMEPath `role_attribute_path`

**Comment by copolycube** [comment]
Hello, @marefr  is this mapping supposed to work with gitlab provider ?
I saw your contribution on https://github.com/grafana/grafana/issues/20735 (okta) and was wondering if this would apply to other OAuth provides too ?

Thanks !

**Comment by copolycube** [comment]
I think that I was using a too old version : https://github.com/grafana/grafana/pull/52407 has been merged  and should be present since https://github.com/grafana/grafana/releases/tag/v9.1.0-beta1

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
